filters/builtin/timeout.go

```diff
@@ -6,22 +6,56 @@ import (
 	"github.com/zalando/skipper/filters"
 )
 
+type timeoutType int
+
+const (
+	backendTimeout timeoutType = iota + 1
+	readTimeout
+	writeTimeout
+)
+
 type timeout struct {
+	typ     timeoutType
```

Comment:
I guess we could use string filter name here which would not require a new type and would simplify Name():
```
func (t *timeout) Name() string {
  return string(t.name)
}
```